main.go

```diff
@@ -56,52 +43,44 @@ func main() {
 		klog.Fatalf("Error building K8s client: %s", err.Error())
 	}
 
-	portClient, err := cli.New(portBaseURL,
-		cli.WithClientID(portClientId), cli.WithClientSecret(portClientSecret),
-		cli.WithDeleteDependents(deleteDependents), cli.WithCreateMissingRelatedEntities(createMissingRelatedEntities),
-		cli.WithHeader("User-Agent", fmt.Sprintf("port-k8s-exporter/0.1 (statekey/%s)", stateKey)),
+	portClient, err := cli.New(config.ApplicationConfig.PortBaseURL,
+		cli.WithClientID(config.ApplicationConfig.PortClientId), cli.WithClientSecret(config.ApplicationConfig.PortClientSecret),
+		cli.WithHeader("User-Agent", fmt.Sprintf("port-k8s-exporter/0.1 (statekey/%s)", config.ApplicationConfig.StateKey)),
 	)
 
 	if err != nil {
 		klog.Fatalf("Error building Port client: %s", err.Error())
 	}
 
-	exporterConfig, err := config.New(configFilePath, resyncInterval, stateKey, eventListenerType)
+	exporterConfig, err := config.GetConfigFile(config.ApplicationConfig.ConfigFilePath, config.ApplicationConfig.ResyncInterval, config.ApplicationConfig.StateKey, config.ApplicationConfig.EventListenerType)
 	if err != nil {
```

Comment:
That means that you have to have a config file even though you can have an integration instead, no?

Reply:
Good catch now im not failing right away ill first check if there is an integration config